src/app.js

```diff
@@ -184,56 +177,57 @@ var run = function () {
   var editor = new Editor(document.getElementById('input'))
 
   // ---------------- FilePanel --------------------
+  // TODO: All FilePanel related CSS should move into file-panel.js
+  // app.js provides file-panel.js with a css selector or a DOM element
+  // and file-panel.js adds its elements (including css), see "Editor" above
   var css = csjs`
     .filepanel    {
       display     : flex;
       width       : 200px;
     }
   `
-  var filepanel = document.querySelector('#filepanel')
-  filepanel.className = css.filepanel
+  var filepanelContainer = document.querySelector('#filepanel')
+  filepanelContainer.className = css.filepanel
   var FilePanelAPI = {
     createName: createNonClashingName,
     switchToFile: switchToFile,
     event: this.event
   }
-  var el = new FilePanel(FilePanelAPI, files)
-  filepanel.appendChild(el)
-  var api = el.api
+  var filePanel = new FilePanel(FilePanelAPI, files)
+  // TODO this should happen inside file-panel.js
+  filepanelContainer.appendChild(filePanel)
 
-  api.register('ui', function changeLayout (data) {
+  filePanel.events.register('ui-hidden', function changeLayout (isHidden) {
     var value
-    if (data.type === 'minimize') {
+    if (isHidden) {
       value = -parseInt(window['filepanel'].style.width)
       value = (isNaN(value) ? -window['filepanel'].getBoundingClientRect().width : value)
       window['filepanel'].style.position = 'absolute'
       window['filepanel'].style.left = (value - 5) + 'px'
       window['filepanel'].style.width = -value + 'px'
       window['tabs-bar'].style.left = '45px'
-    } else if (data.type === 'maximize') {
+    } else {
       value = -parseInt(window['filepanel'].style.left) + 'px'
       window['filepanel'].style.position = 'static'
       window['filepanel'].style.width = value
       window['filepanel'].style.left = ''
       window['tabs-bar'].style.left = value
-    } else {
-      window['filepanel'].style.width = data.width + 'px'
-      window['tabs-bar'].style.left = data.width + 'px'
     }
   })
-  api.register('focus', function (path) {
-    [...window.files.querySelectorAll('.file .name')].forEach(function (span) {
-      if (span.innerText === path) switchToFile(path)
-    })
+  filePanel.events.register('ui-resize', function changeLayout (width) {
+    window['filepanel'].style.width = width + 'px'
+    window['tabs-bar'].style.left = width + 'px'
   })
   files.event.register('fileRenamed', function (oldName, newName) {
+    // TODO please never use 'window' when it is possible to use a variable
+    // that references the DOM node
```

Comment:
I understand.
Basically this comes from the very bad habit of using ID's in html elements.
Every time somebody uses an ID for an html element, like for example `<div id="foobar">...</div>`, the **global variable** `window.foobar` will automatically be created and the DOM element with that ID "foobar" will be assigned.

So I did not bother to create a lof of variables, like for example:
`var filesTabBar = document.querySelector('#files')` because there is already the automatically created global variable `window.files` that references the "tab bar".

I would highly recommend to slowly refactor the code base away from any uses of "ID"s in html tags :-)